### PR TITLE
Fix instructor notes 505

### LINKED ISSUE #505: Bug: XML Parsing error if instructor notes are empty
As far as I understand, the `instructors/instructor-notes.md` provide the information at the top, while the rest of the notes is auto-generated.

Currently you HAVE to write something there, i.e.

```instructor-notes.md
---
title: Instructor notes
---

```

will fail with an error:

```
Fehler in UseMethod("xml_find_all") : 
  nicht anwendbare Methode für 'xml_find_all' auf Objekt der Klasse "xml_document" angewendet
```

using [sandpaper (0.12.4)](https://github.com/carpentries/sandpaper/tree/0.12.4), [pegboard (0.5.3)](https://github.com/carpentries/pegboard/tree/0.5.3), and [varnish (0.2.18)](https://github.com/carpentries/varnish/tree/0.2.18).

### PR DESCRIPTION
**Pull request by klbarnes20**
This PR attempts to fix #505. It adds a test for when instructor notes is empty and an else clause to the build_instructor_notes.R. 

When I ran the test before the fix, it failed. When I ran it after, that particular test passed. I still have two other failed tests, but not sure they are related to this issue/fix.

### CODE REVIEW

**Comment by zkamvar** [comment]
Thank you for submitting the fix, Kelly! 

> When I ran the test before the fix, it failed. When I ran it after, that particular test passed. I still have two other failed tests, but not sure they are related to this issue/fix.

The other two failed tests have to do with the fact that [533d019](https://github.com/carpentries/sandpaper/pull/509/commits/533d01920a7aec709d04ff601ac8174fd82ea350) accidentally deleted a snapshot test (probably because it was skipped on windows). Now that pandoc is more reliably set up for windows, it may be worthwhile revisiting the tests that skip on windows for pandoc, but for now, i will reinstate that snapshot

File: tests/testthat/test-build_lesson.R [hunk 1]
```diff
@@ -165,6 +165,36 @@ test_that("instructor-notes page can be rebuilt", {
...
+    readLines(fs::path(sitepath, "instructor-notes.html"))
+  )))
+  # 1. make a copy of the instructor-notes.md to a local tempfile (use tmp <-
+  # withr::local_tempfile() and fs::file_copy()
+  tmp <- withr::local_tempfile()
+  fs::file_copy(fs::path(res, "instructors/instructor-notes.md"), tmp)
+  # 2. use withr::defer() to do the opposite, copying over the saved file back
+  # when the test finishes)
+  withr::defer({
+    fs::file_copy(tmp, fs::path(res, "instructors/instructor-notes.md"))
```
Comment (zkamvar) [inline tests/testthat/test-build_lesson.R:183]: It looks like we forgot to add the `overwrite = TRUE` argument (which you DID include in your first draft, so my bad), which is causing some problems:

```suggestion
    fs::file_copy(tmp, fs::path(res, "instructors/instructor-notes.md"), overwrite = TRUE)
```

**Review by zkamvar — COMMENTED** [review]
Thank you, Kelly. I have one suggestion that will likely fix the failing test.

**Comment by klbarnes20** [comment]
Thanks @zkamvar! I've committed your suggestions. If the tests pass and you think its ready, feel free to merge.

**Comment by zkamvar** [comment]
Congratulations, @klbarnes20 you have your first contribution to sandpaper!

**Pull request merged**